modules/nixos/impermenance.nix

```diff
@@ -0,0 +1,61 @@
+{ config
+, lib
+, ...
+}:
+let
+  inherit (lib) mkIf mkEnableOption;
+  cfg = config.custom.impermanence;
+in
+{
+  options.custom.impermanence = {
+    enable = mkEnableOption "impermenance";
+  };
+
+  config = mkIf cfg.enable {
+    boot.initrd.systemd.enable = true;
+    boot.initrd.systemd.services.reset-root = {
```

Comment:
Could use `mkIf` to determine if we should use a systemd service or `postDeviceCommands`.

```nix
boot.initrd.postDeviceCommands = mkIf (!config.boot.initrd.systemd.enable) (mkBefore '' '');
```